homeassistant/components/bang_olufsen/event.py

```diff
@@ -71,8 +150,55 @@ async def async_added_to_hass(self) -> None:
             )
         )
 
-    @callback
-    def _async_handle_event(self, event: str) -> None:
-        """Handle event."""
-        self._trigger_event(event)
-        self.async_write_ha_state()
+
+class BangOlufsenRemoteKeyEvent(BangOlufsenEvent):
+    """Event class for Beoremote One key events."""
+
+    _attr_event_types = BEO_REMOTE_KEY_EVENTS
+
+    def __init__(
+        self,
+        config_entry: BangOlufsenConfigEntry,
+        remote: PairedRemote,
+        key_type: str,
+    ) -> None:
+        """Initialize Beoremote One key."""
+        super().__init__(config_entry)
+
+        if TYPE_CHECKING:
+            assert remote.serial_number
+
+        self._attr_unique_id = (
+            f"{remote.serial_number}_{config_entry.unique_id}_{key_type}"
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{remote.serial_number}_{config_entry.unique_id}")},
+            name=f"{BangOlufsenModel.BEOREMOTE_ONE}-{remote.serial_number}-{config_entry.unique_id}",
+            model=BangOlufsenModel.BEOREMOTE_ONE,
+            serial_number=remote.serial_number,
+            sw_version=remote.app_version,
+            manufacturer=MANUFACTURER,
+            via_device=(DOMAIN, self._unique_id),
+        )
+
+        # Make the native key name Home Assistant compatible
+        self._attr_translation_key = key_type.lower().replace("/", "_")
+
+        self._key_type = key_type
+
+    async def async_added_to_hass(self) -> None:
+        """Listen to WebSocket Beoremote One key events."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{self._unique_id}_{CONNECTION_STATUS}",
```

Comment:
The dispatcher is global for all of Home Assistant. We always need to prefix the signal key string with the domain name of the integration that uses the dispatcher.